src/audit/cache_poisoning.rs

```diff
@@ -218,31 +265,18 @@ impl CachePoisoning {
             }
         }
     }
-}
-
-impl WorkflowAudit for CachePoisoning {
-    fn new(_: AuditState) -> anyhow::Result<Self>
-    where
-        Self: Sized,
-    {
-        Ok(Self)
-    }
-
-    fn audit_step<'w>(&self, step: &Step<'w>) -> anyhow::Result<Vec<Finding<'w>>> {
-        let mut findings = vec![];
-
-        let trigger = &step.workflow().on;
-
-        if !self.trigger_used_when_publishing_artifacts(trigger) {
-            return Ok(findings);
-        }
 
+    fn uses_cache_aware_step<'w>(
+        &self,
+        step: &Step<'w>,
+        scenario: &PublishingArtifactsScenario<'w>,
+    ) -> anyhow::Result<Finding<'w>> {
```

Comment:
It looks like we never use the `Err(...)` variants from this, so maybe this could be `Option<Finding<'w>>` instead?

Reply:
Totally agree, let me fix that!

Reply:
Done [here](https://github.com/woodruffw/zizmor/pull/338/commits/7c203c8974696f834f800f2a8c7f146df1bf3cde)